diff --git a/mobile-shell/src/MobileShellPlus/Application.manifest b/mobile-shell/src/MobileShellPlus/Application.manifest
index 4f789ca..c7f0834 100644
--- a/mobile-shell/src/MobileShellPlus/Application.manifest
+++ b/mobile-shell/src/MobileShellPlus/Application.manifest
@@ -4,7 +4,6 @@
     manifestVersion="1.0">
   <assemblyIdentity
     name="ADeltaX.MobileShell"
-    processorArchitecture="amd64"
     version="5.1.0.0"
     type="win32"/>
   <description>MobileShell</description>
@@ -17,7 +16,7 @@
   </trustInfo>
   <compatibility xmlns="urn:schemas-microsoft-com:compatibility.v1">
     <application>
-      <maxversiontested Id="10.0.18362.0"/>
+      <maxversiontested Id="10.0.26100.0"/>
       <supportedOS Id="{8e0f7a12-bfb3-4fe8-b9a5-48fd50a15a9a}" />
     </application>
   </compatibility>
diff --git a/mobile-shell/src/MobileShellPlus/Utils.cpp b/mobile-shell/src/MobileShellPlus/Utils.cpp
index 250934d..60df796 100644
--- a/mobile-shell/src/MobileShellPlus/Utils.cpp
+++ b/mobile-shell/src/MobileShellPlus/Utils.cpp
@@ -4,8 +4,10 @@
 #include "Utils.h"
 #include <ShellScalingApi.h>
 #include <Winuser.h>
+#include <winternl.h>
 
 #pragma comment(lib, "shell32")
+#pragma comment(lib, "ntdll")
 
 void Utils::SendKeyStroke(WORD key)
 {
@@ -276,7 +278,7 @@ void Utils::SetWinTaskbarPosition(WinTaskbarPosition pos)
 {
 	//Issue: this will change the taskbar position in ALL MONITORS
 	//I wasn't able to find how to change on primary taskbar only, unfortunately
-	//ALSO this makes the rotation a lot slower (because APPBAR & GDI slowness technology™)
+	//ALSO this makes the rotation a lot slower (because APPBAR & GDI slowness technologyï¿½)
 	//
 	//Should we enable this?
 
@@ -400,30 +402,31 @@ bool Utils::SystemUsesLightTheme()
 
 bool Utils::IsWindows11()
 {
-	OSVERSIONINFOW VersionInformation;
+	RTL_OSVERSIONINFOW VersionInformation;
+	VersionInformation.dwOSVersionInfoSize = sizeof(RTL_OSVERSIONINFOW);
 
-	VersionInformation.dwOSVersionInfoSize = sizeof(OSVERSIONINFOW);
-	BOOL result = GetVersionExW(&VersionInformation);
-
-	if (result)
+	if (RtlGetVersion(&VersionInformation) == 0)
 	{
 		return VersionInformation.dwBuildNumber >= 21990;
 	}
 
-	return result;
+	return false;
 }
 
-bool Utils::IsMobileCellularSupported()
+bool Utils::IsSlateMode()
 {
-	OSVERSIONINFOW VersionInformation;
+	return GetSystemMetrics(SM_CONVERTIBLESLATEMODE) == 0;
+}
 
-	VersionInformation.dwOSVersionInfoSize = sizeof(OSVERSIONINFOW);
-	BOOL result = GetVersionExW(&VersionInformation);
+bool Utils::IsMobileCellularSupported()
+{
+	RTL_OSVERSIONINFOW VersionInformation;
+	VersionInformation.dwOSVersionInfoSize = sizeof(RTL_OSVERSIONINFOW);
 
-	if (result)
+	if (RtlGetVersion(&VersionInformation) == 0)
 	{
 		return VersionInformation.dwBuildNumber <= 18908;
 	}
 
-	return result;
+	return true;
 }
\ No newline at end of file
diff --git a/mobile-shell/src/MobileShellPlus/Utils.h b/mobile-shell/src/MobileShellPlus/Utils.h
index c45c2d3..449013d 100644
--- a/mobile-shell/src/MobileShellPlus/Utils.h
+++ b/mobile-shell/src/MobileShellPlus/Utils.h
@@ -45,6 +45,7 @@ public:
 	static void SetWinTaskbarIcons(WinTaskbarIconSize size);
 	static void SetWinTaskbarSize(int area, int size);
 	static bool IsWindows11();
+	static bool IsSlateMode();
 	static bool IsMobileCellularSupported();
 };
 
